fvm/programs/invalidator_test.go

```diff
@@ -47,54 +47,54 @@ func TestModifiedSetsInvalidator(t *testing.T) {
 		},
 	}
 
-	require.True(t, invalidator.ShouldInvalidatePrograms())
+	require.True(t, invalidator.ShouldInvalidateItems())
 	require.True(t, invalidator.ShouldInvalidateEntry(ProgramEntry{}))
 }
 
 func TestChainedInvalidator(t *testing.T) {
-	var chain chainedInvalidators
-	require.False(t, chain.ShouldInvalidatePrograms())
+	var chain chainedDerivedDataInvalidators[ProgramEntry]
+	require.False(t, chain.ShouldInvalidateItems())
 	require.False(t, chain.ShouldInvalidateEntry(ProgramEntry{}))
 
-	chain = chainedInvalidators{}
-	require.False(t, chain.ShouldInvalidatePrograms())
+	chain = chainedDerivedDataInvalidators[ProgramEntry]{}
+	require.False(t, chain.ShouldInvalidateItems())
 	require.False(t, chain.ShouldInvalidateEntry(ProgramEntry{}))
 
-	chain = chainedInvalidators{
+	chain = chainedDerivedDataInvalidators[ProgramEntry]{
 		{
-			Invalidator:   testInvalidator{},
-			executionTime: 1,
+			DerivedDataInvalidator: testInvalidator{},
+			executionTime:          1,
 		},
 		{
-			Invalidator:   testInvalidator{},
-			executionTime: 2,
+			DerivedDataInvalidator: testInvalidator{},
+			executionTime:          2,
 		},
 		{
-			Invalidator:   testInvalidator{},
-			executionTime: 3,
+			DerivedDataInvalidator: testInvalidator{},
+			executionTime:          3,
 		},
 	}
-	require.False(t, chain.ShouldInvalidatePrograms())
+	require.False(t, chain.ShouldInvalidateItems())
 
-	chain = chainedInvalidators{
+	chain = chainedDerivedDataInvalidators[ProgramEntry]{
 		{
-			Invalidator:   testInvalidator{invalidateName: "1"},
-			executionTime: 1,
+			DerivedDataInvalidator: testInvalidator{invalidateName: "1"},
+			executionTime:          1,
 		},
 		{
-			Invalidator:   testInvalidator{invalidateName: "3a"},
-			executionTime: 3,
+			DerivedDataInvalidator: testInvalidator{invalidateName: "3a"},
+			executionTime:          3,
 		},
 		{
-			Invalidator:   testInvalidator{invalidateName: "3b"},
-			executionTime: 3,
+			DerivedDataInvalidator: testInvalidator{invalidateName: "3b"},
+			executionTime:          3,
 		},
 		{
-			Invalidator:   testInvalidator{invalidateName: "7"},
-			executionTime: 7,
+			DerivedDataInvalidator: testInvalidator{invalidateName: "7"},
+			executionTime:          7,
 		},
 	}
-	require.True(t, chain.ShouldInvalidatePrograms())
+	require.True(t, chain.ShouldInvalidateItems())
```

Comment:
o, for consistency, maybe rename ShouldInvalidateItems to ShouldInvalidateEntries